compat/win32/dirent.h

```diff
@@ -1,6 +1,9 @@
 #ifndef DIRENT_H
 #define DIRENT_H
 
+#undef MAX_PATH
+#define MAX_PATH 4096
```

Comment:
Did you add the comment into the sources?

For what i remember this magic number is MAX_PATH(Linux)
And since Git is developed as a helper tool for Linux (in its strict meaning of the kernel) development, its internals are made accounting for such paths. Thus attempts to use full path length Windows can offer would most probably screw the whole program